File: api/csharp/ToolGood.TextFilter.Api/Interfaces/ITextFilterAsyncProvider.cs
#if Async
using System.Threading.Tasks;
using ToolGood.TextFilter.Api.Datas;
using ToolGood.TextFilter.Api.Datas.Requests;

namespace ToolGood.TextFilter.Api.Interfaces
{
    public interface ITextFilterAsyncProvider
    {
        /// <summary>
        /// 文本检测-Text格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult TextFilter(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本检测-Text格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> TextFilterAsync(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本替换-Text格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult TextReplace(TextReplaceAsyncRequest request);
        /// <summary>
        /// 文本替换-Text格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> TextReplaceAsync(TextReplaceAsyncRequest request);

        /// <summary>
        /// 文本检测-Html格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult HtmlFilter(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本检测-Html格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> HtmlFilterAsync(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本替换-Html格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult HtmlReplace(TextReplaceAsyncRequest request);
        /// <summary>
        /// 文本替换-Html格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> HtmlReplaceAsync(TextReplaceAsyncRequest request);

        /// <summary>
        /// 文本检测-Json格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult JsonFilter(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本检测-Json格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> JsonFilterAsync(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本替换-Json格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult JsonReplace(TextReplaceAsyncRequest request);
        /// <summary>
        /// 文本替换-Json格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> JsonReplaceAsync(TextReplaceAsyncRequest request);

        /// <summary>
        /// 文本检测-Markdown格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult MarkdownFilter(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本检测-Markdown格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> MarkdownFilterAsync(TextFilterAsyncRequest request);
        /// <summary>
        /// 文本替换-Markdown格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        CommonResult MarkdownReplace(TextReplaceAsyncRequest request);
        /// <summary>
        /// 文本替换-Markdown格式-异步
        /// </summary>
        /// <param name="request">参数</param>
        /// <returns></returns>
        Task<CommonResult> MarkdownReplaceAsync(TextReplaceAsyncRequest request);




    }

}

#endif

File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Interfaces/ITextFilterAsyncProvider.java
package toolgood.textfilter.api.Interfaces;

import toolgood.textfilter.api.Datas.CommonResult;
import toolgood.textfilter.api.Datas.Requests.TextFilterAsyncRequest;
import toolgood.textfilter.api.Datas.Requests.TextReplaceAsyncRequest;

public interface ITextFilterAsyncProvider {

        /**
         * 文本检测-Text格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult TextFilter(TextFilterAsyncRequest request);

        /**
         * 文本替换-Text格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult TextReplace(TextReplaceAsyncRequest request);

        /**
         * 文本检测-Html格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult HtmlFilter(TextFilterAsyncRequest request);

        /**
         * 文本替换-Html格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult HtmlReplace(TextReplaceAsyncRequest request);

        /**
         * 文本检测-Json格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult JsonFilter(TextFilterAsyncRequest request);

        /**
         * 文本替换-Json格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult JsonReplace(TextReplaceAsyncRequest request);

        /**
         * 文本检测-Markdown格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult MarkdownFilter(TextFilterAsyncRequest request);

        /**
         * 文本替换-Markdown格式-异步
         * 
         * @param request 参数
         * @return
         */
        CommonResult MarkdownReplace(TextReplaceAsyncRequest request);

}
